nginx.conf

```diff
@@ -73,9 +73,9 @@ http
     }
 
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-    proxy_connect_timeout       5s;
-    proxy_send_timeout          5s;
-    proxy_read_timeout          5s;
+    proxy_connect_timeout       15s;
+    proxy_send_timeout          15s;
+    proxy_read_timeout          15s;
```

Comment:
Adjusted binarySearch is too slow for fetching some IPs. The problem is that ranges overlap.